contracts/src/v0.8/functions/tests/1_0_0/Setup.t.sol

```diff
@@ -218,184 +260,237 @@ contract FunctionsClientRequestSetup is FunctionsSubscriptionSetup {
+    signerPrivateKeys[1] = NOP_SIGNER_PRIVATE_KEY_2;
+    signerPrivateKeys[2] = NOP_SIGNER_PRIVATE_KEY_3;
+    (bytes32[] memory rawRs, bytes32[] memory rawSs, bytes32 rawVs) = _signReport(
+      report,
+      reportContext,
+      signerPrivateKeys
     );
-    s_requestCommitment2 = commitment2;
 
-    // Transmit as transmitter 2
+    // Send as transmitter
     vm.stopPrank();
-    vm.startPrank(NOP_TRANSMITTER_ADDRESS_2);
-
-    // Build report
-    bytes32[] memory requestIds2 = new bytes32[](1);
-    requestIds2[0] = s_requestId2;
-    bytes[] memory results2 = new bytes[](1);
-    results2[0] = bytes("hello world!");
-    bytes[] memory errors2 = new bytes[](1);
-    // No error
-    bytes[] memory onchainMetadata2 = new bytes[](1);
-    onchainMetadata2[0] = abi.encode(s_requestCommitment2);
-    bytes[] memory offchainMetadata2 = new bytes[](1);
-    // No offchain metadata
-    bytes memory report2 = abi.encode(requestIds2, results2, errors2, onchainMetadata2, offchainMetadata2);
-
-    // Build signers
-    address[31] memory signers2;
-    signers2[0] = NOP_SIGNER_ADDRESS_2;
+    vm.startPrank(transmitter);
 
     // Send report
     vm.recordLogs();
-    s_functionsCoordinator.callReportWithSigners(report2, signers2);
-
-    // Get actual cost from RequestProcessed event log
-    Vm.Log[] memory entriesAfterFulfill2 = vm.getRecordedLogs();
-    (uint96 totalCostJuels2, , , , , ) = abi.decode(
-      entriesAfterFulfill2[2].data,
-      (uint96, address, FunctionsResponse.FulfillResult, bytes, bytes, bytes)
-    );
-    // totalCostJuels = costWithoutCallbackJuels + adminFee + callbackGasCostJuels
-    s_fulfillmentCoordinatorBalance += totalCostJuels2 - s_adminFee;
-    s_fulfillmentRouterOwnerBalance += s_adminFee;
+    if (transmitterGasToUse > 0) {
+      s_functionsCoordinator.transmit{gas: transmitterGasToUse}(reportContext, report, rawRs, rawSs, rawVs);
+    } else {
+      s_functionsCoordinator.transmit(reportContext, report, rawRs, rawSs, rawVs);
+    }
+
+    if (expectedToSucceed) {
+      // Get actual cost from RequestProcessed event log
+      (uint96 totalCostJuels, , , , , ) = abi.decode(
+        vm.getRecordedLogs()[requestProcessedIndex].data,
+        (uint96, address, FunctionsResponse.FulfillResult, bytes, bytes, bytes)
+      );
+      // Store profit amounts
+      s_fulfillmentRouterOwnerBalance += s_adminFee;
+      // totalCostJuels = costWithoutCallbackJuels + adminFee + callbackGasCostJuels
+      s_fulfillmentCoordinatorBalance += totalCostJuels - s_adminFee;
+    }
 
     // Return prank to Owner
     vm.stopPrank();
     vm.startPrank(OWNER_ADDRESS);
+  }
 
-    //  *** Request #3 ***
-    vm.recordLogs();
-    s_requestId3 = s_functionsClient.sendRequest(
-      s_donId,
-      "return 'hello world';",
-      new bytes(0),
-      new string[](0),
-      new bytes[](0),
-      s_subscriptionId,
-      5500
-    );
+  /// @dev Overload to give transmitterGasToUse as 0, which gives default tx gas
```

Comment:
This seems like quite a lot of repeated code. Is there a better way to structure this?

Reply:
Can we have a single _reportAndStore and just pass the required args every time?

Reply:
> a lot of repeated code
Like what? The only repeated code I see is the fulfillment data, but I think that's better to not re-use

Reply:
> Can we have a single _reportAndStore and just pass the required args every time?

This will make it harder to read & write the test case itself.
If the overloaded function is used, then the args are filled in by it aren't relevant to the test.

I'm open to making this change though if you think it will make it easier to grok

Reply:
Yes that would help.  I just want to make sure I able to grok everything during review and I got a bit confused during the overloaded funcs